File: Julia-files/19_databases.jl
# Requires packages: import Pkg; Pkg.add(["SQLite", "DataFrames"])
using SQLite, DataFrames

# 1. Connect to in-memory database
db = SQLite.DB() 

# 2. Create table and insert data
SQLite.execute(db, "CREATE TABLE data_table (val INTEGER, cat TEXT)")
SQLite.execute(db, "INSERT INTO data_table VALUES (1, 'A')")
SQLite.execute(db, "INSERT INTO data_table VALUES (5, 'C')")

# 3. Execute SQL Query
res = DBInterface.execute(db, "SELECT * FROM data_table WHERE val > 2") |> DataFrame
println(res)

File: R-files/19_databases.R
# Install packages if missing: install.packages(c("DBI", "RSQLite"))
library(DBI)
library(RSQLite)

# 1. Connect to an in-memory database
con <- dbConnect(RSQLite::SQLite(), ":memory:")

# 2. Write sample data to DB
data(mtcars)
dbWriteTable(con, "cars", mtcars)

# 3. Execute SQL Query
res <- dbGetQuery(con, "SELECT mpg, cyl, hp FROM cars WHERE mpg > 20")
print(head(res))

# 4. Disconnect
dbDisconnect(con)